src/screens/WasteCollectionSettingsScreen.tsx

```diff
@@ -266,20 +273,6 @@ export const WasteCollectionSettingsScreen = () => {
     asyncLoadStoredSettingsFromServer();
   }, [typeSettings]);
 
-  // Use this ref to prevent the useEffect from running multiple times
-  const hasStoredStreetSettings = useRef(false);
-
-  useEffect(() => {
-    if (
-      !hasStoredStreetSettings.current &&
-      loadedStoredSettingsInitially &&
-      waste?.streetId !== selectedStreetId
-    ) {
-      setSelectedStreetId(selectedStreetId);
-      hasStoredStreetSettings.current = true;
-    }
-  }, [loadedStoredSettingsInitially, waste.streetId, selectedStreetId]);
```

Comment:
`setSelectedStreetId(selectedStreetId);` made no sense. i think this condition never evaluated to true. but if it would, the state would be set with the current value of the state? strange. i did not encounter any issues after removal.